packages/fx-core/src/plugins/resource/sql/plugin.ts

```diff
@@ -83,70 +84,26 @@ export class SqlPluginImpl {
     stage: Stage,
     ctx: PluginContext
   ): Promise<Result<QTreeNode | undefined, FxError>> {
-    if (stage === Stage.provision) {
-      ctx.logProvider?.info(Message.startGetQuestions);
-      const sqlNode = new QTreeNode({
-        type: "group",
-      });
-      if (ctx.answers?.platform === Platform.CLI_HELP) {
-        this.buildQuestionNode(sqlNode);
-        return ok(sqlNode);
-      }
-
-      await this.init(ctx);
-      this.config.prepareQuestions = true;
-      if (isArmSupportEnabled()) {
-        this.config.admin = ctx.config.get(Constants.admin) as string;
-        this.config.adminPassword = ctx.config.get(Constants.adminPassword) as string;
-        if (this.config.admin) {
-          this.config.existSql = true;
-        }
-      } else if (this.config.azureSubscriptionId) {
-        const managementClient: ManagementClient = await ManagementClient.create(ctx, this.config);
-        this.config.existSql = await managementClient.existAzureSQL();
-      }
-
-      if (!this.config.existSql) {
-        this.buildQuestionNode(sqlNode);
-      }
+    if (stage === Stage.provision && ctx.answers?.platform === Platform.CLI_HELP) {
+      const sqlNode = this.buildQuestionNode();
       return ok(sqlNode);
     }
     return ok(undefined);
   }
 
   async preProvision(ctx: PluginContext): Promise<Result<any, FxError>> {
     ctx.logProvider?.info(Message.startPreProvision);
-
     await this.init(ctx);
-    if (!this.config.azureSubscriptionId) {
-      const error = SqlResultFactory.SystemError(
-        ErrorMessage.SqlGetConfigError.name,
-        ErrorMessage.SqlGetConfigError.message(
-          Constants.solutionConfigKey.subscriptionId,
-          Constants.solution
-        )
-      );
-      ctx.logProvider?.error(error.message);
-    }
 
     DialogUtils.init(ctx);
     TelemetryUtils.init(ctx);
     TelemetryUtils.sendEvent(Telemetry.stage.preProvision + Telemetry.startSuffix);
 
-    const skipAddingUser = ctx.config.get(Constants.skipAddingUser);
-    if (skipAddingUser === undefined) {
-      this.config.skipAddingUser = (await ctx.azureAccountProvider?.getIdentityCredentialAsync())
-        ? false
-        : true;
-    } else {
-      this.config.skipAddingUser = skipAddingUser as boolean;
-    }
-
-    if (!this.config.prepareQuestions) {
-      this.config.existSql = true;
-    }
+    await this.loadSkipAddingUser(ctx);
+    await this.checkSqlExisting(ctx);
 
     if (!this.config.existSql) {
+      await this.askInputs(ctx);
```

Comment:
so we still only ask password when admin not exists? So in re-provision with rg deleted, question is not asked?

Reply:
In no-arm support scenario,  the flag config.existSql depends on checking sql endpoint. It will ask password if check endpoint failed. It works well in re-provision regardless rg deleted or not.
In arm support scenario, the flag config.existSql depends on the admin existing. The admin will put in config once provision succeeded.
Currently it will fail for creating new sql instance if admin is existing and rg is deleted, since password is required but we store admin but not password.